### PR TITLE
test: net: net and conngater tests

### PR DESCRIPTION
**Pull request by TheMenko**
## Related Issues
-

## Proposed Changes
This PR adds network integration tests specifically from node net and conngater files.
It has annotations that system test matrix uses (//stm comments) 

## Additional Info
-

## Checklist

Before you mark the PR ready for review, please make sure that:
- [x] All commits have a clear commit message.
- [x] The PR title is in the form of of `<PR type>: <area>: <change being made>`
    - example: ` fix: mempool: Introduce a cache for valid signatures`
    - `PR type`: _fix_, _feat_, _INTERFACE BREAKING CHANGE_, _CONSENSUS BREAKING_, _build_, _chore_, _ci_, _docs_,_perf_, _refactor_, _revert_, _style_, _test_
    - `area`: _api_, _chain_, _state_, _vm_, _data transfer_, _market_, _mempool_, _message_, _block production_, _multisig_, _networking_, _paychan_, _proving_, _sealing_, _wallet_, _deps_
- [x] This PR has tests for new functionality or change in behaviour
- [x] If new user-facing features are introduced, clear usage guidelines and / or documentation updates should be included in https://lotus.filecoin.io or [Discussion Tutorials.](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] CI is green

### CODE REVIEW

**Review by TheDivic — COMMENTED** [review]
Make sure to check if this is not flaky by running it with `go test -count=N`!

File: itests/net_test.go [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+	bandwidth, err := secondNode.NetBandwidthStats(ctx)
+	require.NoError(t, err)
+
+	//stm: @NETWORK_COMMON_BANDWIDTH_STATS_BY_PEER_001
+	peerBandwidths, err := firstNode.NetBandwidthStatsByPeer(ctx)
+	require.NoError(t, err)
+
+	if peerBandwidths[secondNodeID.String()] != bandwidth {
+		t.Errorf("bandwidths differ")
+	}
```
Comment (magik6k) [inline itests/net_test.go:65]: Those can be different, and likely will be when non-zero values get reported

File: itests/net_test.go [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+
+	if connState != network.Connected {
+		t.Errorf("peer does not have connected state")
+	}
+
+	//stm: @NETWORK_COMMON_PEERS_001
+	peers, err := firstNode.NetPeers(ctx)
+	require.NoError(t, err)
+
+	if len(peers) > 0 && peers[0].ID != addrInfo.ID {
```
Comment (magik6k) [inline itests/net_test.go:85]: I'd check peer count with `== 1`

File: itests/net_test.go [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+	err := firstNode.NetBlockAdd(ctx, api.NetBlockList{IPAddrs: secondNodeIPs})
+	require.NoError(t, err)
+
+	//stm: @NETWORK_COMMON_BLOCK_LIST_001
+	list, err := firstNode.NetBlockList(ctx)
+	require.NoError(t, err)
+
+	if len(list.IPAddrs) == 0 || list.IPAddrs[0] != secondNodeIPs[0] {
+		t.Errorf("blocked ip not in blocked ip list")
+	}
```
Comment (magik6k) [inline itests/net_test.go:235]: Can we try to NetConnect here, and see if it fails?

Comment (TheMenko) [reply itests/net_test.go:235]: Hey @magik6k , I asked about this part on slack but it got deleted, i'm not sure how this should work.
NetConnect won't fail there, nodes will still be able to connect, and that is true for peer,subnet and ip blocking.
Besides that, i tried to use full node and miner node, then block miner node and try to mine block and sync with full node, and even that works. 
So blocking a node seems to not do anything. Do you have any tips on how this should be tested?

File: itests/kit/ensemble.go [hunk 4]
```diff
@@ -340,7 +340,6 @@ func (n *Ensemble) Start() *Ensemble {
 			node.FullAPI(&full.FullNode, node.Lite(full.options.lite)),
 			node.Base(),
 			node.Repo(r),
-			node.MockHost(n.mn),
```
Comment (arajasek) [inline itests/kit/ensemble.go:343]: I'm not sure what the impact of using a real host for all tests is...if necessary, we can have an enablep2p option just for the net tests.

File: itests/net_test.go (hunk 3 above)
Comment (arajasek) [reply itests/net_test.go]: Added this check

**Review by arajasek — APPROVED** [review]

**Pull request merged**